pkg/helpers/helpers_test.go

```diff
@@ -231,3 +233,109 @@ func TestEqualError(t *testing.T) {
 		}
 	}
 }
+
+func TestBashEscapeString(t *testing.T) {
+
+	tests := []struct {
+		input    string
+		expected string
+	}{
+		{
+			"Hel'lo'p\"la`ygr'ound'",
+			`'Hel'"'"'lo'"'"'p"la` + "`" + `ygr'"'"'ound'"'"`,
+		},
+		{
+			`"PwEV@QG7/PYt"re9`,
+			`'"PwEV@QG7/PYt"re9'`,
+		},
+		{
+			"",
+			"",
+		},
+		{
+			"plaintext1234",
+			`'plaintext1234'`,
+		},
+		{
+			"Hel'lo'p\"la`ygr'ound",
+			`'Hel'"'"'lo'"'"'p"la` + "`" + `ygr'"'"ound`,
+		},
+		{
+			`conse''cutive`,
+			`'conse'"'"''"'"cutive`,
+		},
+		{
+			"conse\\\\cutive",
+			`'conse\\cutive'`,
+		},
+		{
+			"consec\"\"utive",
+			`'consec""utive'`,
+		},
+		{
+			`PwEV@QG7/PYt"re9`,
+			`'PwEV@QG7/PYt"re9'`,
+		},
+		{
+			"Lnsr@191",
+			"'Lnsr@191'",
+		},
+		{
+			"Jach#321",
+			"'Jach#321'",
+		},
+		{
+			"Bgmo%219",
+			"'Bgmo%219'",
+		},
+		{
+			"@#$%^&*-_!+=[]{}|\\:,.?/~\"();" + "`",
+			`'@#$%^&*-_!+=[]{}|\:,.?/~"();` + "`'",
+		},
+	}
+
+	for _, test := range tests {
+		actual := BashEscapeString(test.input)
+
+		if actual != test.expected {
+			t.Errorf("expected %s but got %s", test.expected, actual)
+		}
+	}
+
+}
+
+func TestBashEscapeStringE2E(t *testing.T) {
+
+	tests := []string{
+		"Hel'lo'p\"la`ygr'ound'",
+		`"PwEV@QG7/PYt"re9`,
+		"",
+		"plaintext1234",
+		"Hel'lo'p\"la`ygr'ound",
+		`conse''cutive`,
+		"conse\\\\cutive",
+		"consec\"\"utive",
+		`PwEV@QG7/PYt"re9`,
+		"Lnsr@191",
+		"Jach#321",
+		"Bgmo%219",
+	}
+
+	for _, test := range tests {
```

Comment:
why not merge the two test functions so we only call `BashEscapeString` once for each test case?

Reply:
Sure thing.